File: ruby-mnp-plasmon/examples/spectrum_csv.rb
require "csv"
require "fileutils"
require_relative "../lib/mnp_plasmon"

wavelengths = (400..700).step(5)
out_path = File.expand_path("../../docs/example-data/ruby.csv", __dir__)
FileUtils.mkdir_p(File.dirname(out_path))

CSV.open(out_path, "w") do |csv|
  csv << %w[wavelength_nm c_ext c_sca c_abs]
  wavelengths.each do |wl|
    result = MnpPlasmon.sphere_response(
      wavelength_nm: wl.to_f,
      radius_nm: 25.0,
      material: "Au",
      medium_refractive_index: 1.33
    )
    csv << [wl, result[:c_ext], result[:c_sca], result[:c_abs]]
  end
end

File: c-mnp-plasmon/examples/spectrum_csv.c
#include "../src/mnp_plasmon.h"
#include <stdio.h>

int main(void) {
    FILE *fp = fopen("docs/example-data/c.csv", "w");
    if (!fp) {
        return 1;
    }

    fprintf(fp, "wavelength_nm,c_ext,c_sca,c_abs\n");
    for (int wl = 400; wl <= 700; wl += 5) {
        sphere_response_t r = mnp_simulate_sphere_response("Au", (double)wl, 25.0, 1.33);
        fprintf(fp, "%d,%.12g,%.12g,%.12g\n", wl, r.c_ext, r.c_sca, r.c_abs);
    }

    fclose(fp);
    return 0;
}
